### DISCUSSION TITLE
Animating over a static image/sprite

### DISCUSSION THREAD

**Discussion by rayegun** [Bodmer/TFT_eSPI#2920] [Q&A - General]
I figured I'd ask if there was a known method to do this before coding it up myself:

I want to have a static background image (or perhaps a sprite?) which I animate shapes over top of (without deleting). The minimum example I want is filling a circle symmetrically from the outside in, and then reversing that. Essentially a blink or an irising effect.

Is that doable? My immediate thought is a derivative of https://github.com/Bodmer/TFT_eSPI/blob/master/examples/320%20x%20240/TFT_ArcFill/TFT_ArcFill.ino but I'm not sure the ideal way to maintain the image without layers (and smoothly, perhaps with DMA).

I have more complex animations I just plan on doing with GIFs but this seems simple enough I could do it programmatically.

**Comment by Bodmer** [comment]
The library does not have any functions specifically for animations. Possibilities include:

1. Using animated GIFs
2. Pushing images to a screen with a mask
3. Pushing sprites to a screen with a mask

For the latter 2 this function can be used:
pushMaskedImage(int32_t x, int32_t y, int32_t w, int32_t h, uint16_t *img, uint8_t *mask);

The mask can be a 1bpp sprite created and drawn in by the sketch.

I have created an example using sprites:
[Draw_masked_sprite.zip](https://github.com/Bodmer/TFT_eSPI/files/13169975/Draw_masked_sprite.zip)

Note there is a bug in the library that will need to be corrected for this to work. The master library has been updated, see here:
https://github.com/Bodmer/TFT_eSPI/issues/2924

You can either add that line in your copy of the library or download the master library direct from github.

This fun example shows how the principle can be used for simple masked animations:
[Snowflakes_v2.zip](https://github.com/Bodmer/TFT_eSPI/files/13170041/Snowflakes_v2.zip)

In this example masks are created out of the letters and this mask is then used on the snowflakes sprite.
At the bottom you can see the snowlakes sprite and the mask containing the letter "D"
